### PR TITLE
TextFieldWidget mappings

### PR DESCRIPTION
**Pull request by TheGlitch76**
I'm fairly certain that `method_1853` `method_1864` and `method_1869` are all just overloaded methods with the same name, but I can name them separately if desired.

### CODE REVIEW

File: mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping [hunk 1]
```diff
@@ -40,31 +43,53 @@ CLASS net/minecraft/class_342 net/minecraft/client/gui/widget/TextFieldWidget
...
+		ARG 2 position2
+		ARG 3 stopAtEndOfWord
 	METHOD method_1865 tick ()V
 	METHOD method_1866 getSelectedText ()Ljava/lang/String;
 	METHOD method_1867 addText (Ljava/lang/String;)V
 	METHOD method_1868 setEditableColor (I)V
 		ARG 1 color
+	METHOD method_1869 getDistanceBetween (II)I
+		ARG 1 position1
+		ARG 2 position2
```
Comment (Sollace) [inline mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping]: `start` and `end`?

File: mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping [hunk 2]
```diff
@@ -40,31 +43,53 @@ CLASS net/minecraft/class_342 net/minecraft/client/gui/widget/TextFieldWidget
...
 	METHOD method_1882 getText ()Ljava/lang/String;
+	METHOD method_1883 setCursorKeepingSelection (I)V
+		ARG 1 cursor
+	METHOD method_1884 setCursorMin (I)V
+		ARG 1 cursorMin
 	METHOD method_1885 isVisible ()Z
+	METHOD method_1886 drawSelectionOverlay (IIII)V
 	METHOD method_1887 setSuggestion (Ljava/lang/String;)V
 		ARG 1 suggestion
 	METHOD method_1888 setIsEditable (Z)V
```
Comment (Sollace) [inline mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping]: I would think the `Is` part here to be a bit redundant.

Comment (TheGlitch76) [reply mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping]: Those mappings were made by Gegy a while ago. I can change them to follow new conventions if breaking changes are OK.

File: mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping [hunk 3]
```diff
@@ -40,40 +43,63 @@ CLASS net/minecraft/class_342 net/minecraft/client/gui/widget/TextFieldWidget
 	METHOD method_1851 hasBorder ()Z
 	METHOD method_1852 setText (Ljava/lang/String;)V
 		ARG 1 text
+	METHOD method_1853 getDistanceBetween (I)I
+		ARG 1 position1
```
Comment (Gegy) [inline mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping]: What is the distance between here? (I imagine some internal field?)

File: mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping [hunk 4]
```diff
@@ -31,7 +31,7 @@ CLASS net/minecraft/class_342 net/minecraft/client/gui/widget/TextFieldWidget
 	METHOD charTyped (CI)Z
 		ARG 1 chr
 		ARG 2 keyCode
-	METHOD isMouseOver (DD)Z
+	METHOD mouseOver (DD)Z
```
Comment (Gegy) [inline mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping]: The getters should stay with `is` iirc (may be wrong, did standard change?)

File: mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping (hunk 3 above)
Comment (Runemoro) [reply mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping]: `getDistanceToCursor(pos)`

File: mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping (hunk 4 above)
Comment (Sollace) [reply mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping]: No. Getters with an `is` are fine.

**Comment by modmuss50** [comment]
Needs suggestions to be addressed please.

**Comment by TheGlitch76** [comment]
Understood. Will take care of it soon.

File: mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping (hunk 4 above)
Comment (shedaniel) [reply mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping]: Is this method supposed to be an override of `Element.isMouseOver`?

Comment (TheGlitch76) [reply mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping]: Fixed

File: mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping (hunk 3 above)
Comment (TheGlitch76) [reply mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping]: Using Runemoro's suggestion instead of treating it as an overload.

**Comment by TheGlitch76** [comment]
Ok. I've taken care of all the reviews.

**Review by modmuss50 — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Comment by TheGlitch76** [comment]
This PR was completely superseded by 22d643e1135a36031bbae4a50f8b3213ae648502 and possibly other commits.

**Comment by Sollace** [comment]
_oof_

**Pull request closed without merging**